lib/compilers/dotnet.ts

```diff
@@ -193,10 +204,53 @@ class DotNetCompiler extends BaseCompiler {
         return compilerResult;
     }
 
-    override optionsForFilter() {
+    override optionsForFilter(filters: ParseFilters) {
+        if (filters.binary) {
+            this.compileToBinary = true;
+        }
+
         return this.compilerOptions;
     }
 
+    override async execBinary(
```

Comment:
this should error for non-default arch or non `--targetarch x64`

Reply:
The --targetarch is used for crossgen2, while execution use corerun to execute the managed assembly directly, which is arch-independent.

Reply:
ah ok.
btw if we pass wrong argument, currently there is no validation and error reporting which crossgen2 provide but compiler-explorer fails to pass-thru https://github.com/compiler-explorer/compiler-explorer/issues/4218 (i had a typo: `--targtarch arm64` and was banging my head why i'm seeing x64 assembly. nothing was on the output.

<img width="426" alt="image" src="https://user-images.githubusercontent.com/83082615/204142839-457e5967-b4a3-4a66-977f-5d59ee993ac6.png">

is this pr fixing that too?

Reply:
No. We use a filter and simply ignore unexpected args for now. 

See: https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/compilers/dotnet.ts#L53-L78

Fixing it is out of scope for this PR.